mk/checkconf.mk

```diff
@@ -95,19 +95,20 @@ define cfg-cmake-set
 		  set($1 $($1))_nl_))
 endef
 
-# Returns 'y' if at least one variable is 'y', empty otherwise
+# Returns 'y' if at least one variable is 'y', 'n' otherwise
 # Example:
 # FOO_OR_BAR := $(call cfg-one-enabled, FOO BAR)
-cfg-one-enabled = $(if $(filter y, $(foreach var,$(1),$($(var)))),y,)
+cfg-one-enabled = $(if $(filter y, $(foreach var,$(1),$($(var)))),y,n)
 
-# Returns 'y' if all variables are 'y', empty otherwise
+# Returns 'y' if all variables are 'y', 'n' otherwise
 # Example:
 # FOO_AND_BAR := $(call cfg-all-enabled, FOO BAR)
 cfg-all-enabled =                                                             \
     $(strip                                                                   \
         $(if $(1),                                                            \
             $(if $(filter y,$($(firstword $(1)))),                            \
                 $(call cfg-all-enabled,$(filter-out $(firstword $(1)),$(1))), \
+		n                                                             \
              ),                                                               \
             y                                                                 \
```

Comment:
By the way, I could find why there's such a `y` at this line.

I would have thought there should rather be a `n` here. Current script leads to a strange behavior when used with unexpected empty like arguments:
`foo := $(call cfg-all-enabled,)` leads to `foo=y`
while `foo := $(call cfg-all-enabled, )` leads to `foo=n`.

Reply:
> By the way, I could not find why there's such a `y` at this line.

Because if we reach this point in means that the list of variables is empty and all have been found to be `y`, so the overall result shall be `y`.

> `foo := $(call cfg-all-enabled,)` leads to `foo=y`

Yes, because of the above. Frankly it's a use case that does not matter much IMO.

> while `foo := $(call cfg-all-enabled, )` leads to `foo=n`

Leads to `foo=` (empty) with the original code. That's because space is not empty so the 'if' we check the value of `$( )` which is not `y` hence returning empty immediately.

OK, all this is a bit confusing, here is what I propose: please undo the changes to `cfg-all-enabled` and keep only the changes to `cfg-one-enabled` as you did initially. I will propose another patch for `cfg-all-enabled` to behave like the new `cfg-one-enabled`, if you don't mind.

Reply:
ok, thanks.